spec/i18n_spec.rb

```diff
@@ -140,15 +140,13 @@ def allowed_untranslated_key?(locale, key)
   end
 end
 
-RSpec.describe 'I18n' do
-  let(:i18n) { I18n::Tasks::BaseTask.new }
-  let(:missing_keys) { i18n.missing_keys }
-  let(:unused_keys) { i18n.unused_keys }
-  let(:untranslated_keys) { i18n.untranslated_keys }
-  let(:leading_or_trailing_whitespace_keys) do
-    i18n.leading_or_trailing_whitespace_keys
-  end
+i18n = I18n::Tasks::BaseTask.new
+missing_keys = i18n.missing_keys
+unused_keys = i18n.unused_keys
+untranslated_keys = i18n.untranslated_keys
+leading_or_trailing_whitespace_keys = i18n.leading_or_trailing_whitespace_keys
```

Comment:
The tradeoff here is that everything will now run, even if I'm only running a single test that doesn't use them. I suspect the one that is used in all of them is `i18n`, whereas the others may only be used in a test or two. Is it possible to realize most of the improvements by only applying these changes for `i18n`?

Reply:
What if we broke the i18n-tasks ones into a separate file? And kept our own custom ones in this file, so we can run them individually?

Reply:
To confirm my suspicion, I ran the individual spec "has matching HTML tags" on both `main` and this branch. I think there may need to be a more precise way to approach this to avoid making the case where a subset of tests are run much slower.

| test              | main_speed                                                                        | speed_up_branch_speed                                                            |
| ---------------------- | --------------------------------------------------------------------------------- | -------------------------------------------------------------------------------- |
| has matching HTML tags | Finished in 0.83141 seconds (files took 2.89 seconds to load). 3.72 seconds total | Finished in 0.51449 seconds (files took 11.84 seconds to load). 12.35 seconds total |


```
mitchellehenke:identity-idp/ (main*) $ rspec spec/i18n_spec.rb:289

Run options: include {:locations=>{"./spec/i18n_spec.rb"=>[289]}}

Randomized with seed 32520

I18n
  has matching HTML tags

Top 0 slowest examples (0 seconds, 0.0% of total time):

Finished in 0.83141 seconds (files took 2.89 seconds to load)
1 example, 0 failures

Randomized with seed 32520

mitchellehenke:identity-idp/ (hmm/i18n-speedup-spike) $ rspec spec/i18n_spec.rb:147

Randomized with seed 48937

I18n
  has matching HTML tags

Top 0 slowest examples (0 seconds, 0.0% of total time):

Finished in 0.51449 seconds (files took 11.84 seconds to load)
1 example, 0 failures